sql/src/main/java/io/crate/analyze/symbol/Aggregations.java

```diff
@@ -22,11 +22,18 @@
 
 package io.crate.analyze.symbol;
 
+import io.crate.analyze.validator.LiteralVisitor;
 import io.crate.metadata.FunctionInfo;
+import io.crate.operation.scalar.arithmetic.ArrayFunction;
+
+import java.util.List;
 
 public final class Aggregations {
 
     private final static AggregationSearcher AGGREGATION_SEARCHER = new AggregationSearcher();
+    private final static AggregationOrScalarSearcher AGGREGATION_OR_SCALAR_SEARCHER = new AggregationOrScalarSearcher();
+    private final static GroupBySymbolMatcher GROUP_BY_MATCHER = new GroupBySymbolMatcher();
+    private final static LiteralVisitor LITERAL_VISITOR = new LiteralVisitor();
```

Comment:
How about moving the LiteralVisitor as a static inner class into Symbols and only expose a single static `allLiterals` (not really accurate, as functions are also valid) - so maybe something like `eagerEvaluatable` method? Or turn it around and have a `containsColumns` method?